prime-router/src/main/kotlin/cli/ProcessDataCommands.kt

```diff
@@ -432,24 +437,39 @@ class ProcessData : CliktCommand(
         }
 
         // Output reports
-        writeReportsToFile(outputReports) { report, format, stream ->
-            val hl7Configuration = Hl7Configuration(
-                nameFormat = getNameFormat(Report.NameFormat.STANDARD),
-                suppressQstForAoe = suppressQstForAoe,
-                receivingApplicationName = receivingApplication,
-                receivingFacilityName = receivingFacility,
-                receivingOrganization = receivingOrganization,
-                receivingApplicationOID = "",
-                receivingFacilityOID = "",
-                messageProfileId = "",
-                useBatchHeaders = format == Report.Format.HL7_BATCH,
-                reportingFacilityId = reportingFacilityId,
-                reportingFacilityName = reportingFacilityName,
-            )
+        writeReportsToFile(outputReports, metadata) { report, format, stream ->
             when (format) {
                 Report.Format.INTERNAL -> csvSerializer.writeInternal(report, stream)
                 Report.Format.CSV -> csvSerializer.write(report, stream)
-                Report.Format.HL7 -> hl7Serializer.write(report, stream, hl7Configuration)
+                Report.Format.HL7 -> {
+                    // create a default hl7 config
+                    val hl7Configuration = Hl7Configuration(
+                        nameFormat = nameFormat ?: "standard",
+                        suppressQstForAoe = suppressQstForAoe,
+                        receivingApplicationName = receivingApplication,
+                        receivingFacilityName = receivingFacility,
+                        receivingOrganization = receivingOrganization,
+                        receivingApplicationOID = "",
+                        receivingFacilityOID = "",
+                        messageProfileId = "",
+                        useBatchHeaders = format == Report.Format.HL7_BATCH,
+                        reportingFacilityId = reportingFacilityId,
+                        reportingFacilityName = reportingFacilityName,
+                    )
+                    // and create a report with a new destination
+                    val reportWithTranslation = if (report.destination?.translation == null) {
+                        val destination = Receiver(
+                            "emptyReceiver",
+                            "emptyOrganization",
+                            "covid-19",
+                            hl7Configuration
+                        )
+                        report.copy(destination, Report.Format.HL7_BATCH)
+                    } else {
+                        report
+                    }
+                    hl7Serializer.write(reportWithTranslation, stream)
+                }
                 Report.Format.HL7_BATCH -> hl7Serializer.writeBatch(report, stream)
```

Comment:
For HL7, I am creating a blank `Receiver` and `Hl7Configuration` for generating file names.